### DISCUSSION TITLE
Testing expansions panel ~ one discussion, likely more than one point...

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1503] [Ideas]
I've already got one issue ticket wrt A2386SX expansion board emulation ~ that's an odd-ball case, in that you've got expansions, to the A2386SX expansion card -- perhaps it deserves it's own ticket...the rest though?...I'm not so sure ~ maybe I can keep notes here, and they can be cherry picked over time (seemed to work wrt GUI theming ;)

I created a baseline A1200 machine type, 2 + 8, CPU set to **fastest possible**. DH0: is a pristine Workbench 3.1 install. Basic benchmarker is AIBB v6.5 ....I decided to check something familiar ~ emulate my A1200, which has a DKB-Cobra 030/882 + Ferret ... as a starting point, A1200 stock (2 +8), no FPU ...

![ex](https://imgur.com/Y4GijNo.png)

....A1200 stock (2 +8), with FPU ...

![ex1](https://imgur.com/l4oNfdc.png)

....A1200 stock (2 +8), with FPU + DKB-1230 (no Ferret)...

![ex2](https://imgur.com/I448jFP.png)

Observation;

There are only 2 valid ROM images for this expansion ~ with or without the Ferret SCSI expansion...

DKB_1230_1240_Cobra_DKBRom.bin
DKB_1230_1240_Cobra_v1.23_DKBSCSIRom.bin

![ex3](https://imgur.com/9IaNpul.png)

At this point, with DKB selected and the SCSI (Ferret) unchecked, the ROM field/dropdown selector (no label) should auto select (if present in the ROMs path) the DKB_1230_1240_Cobra_DKBRom.bin rom image.

Likewise if user also checks the SCSI (Ferret) widget (no idea why one would), it should auto select the other rom image to suit the selected (sub)expansion ... DKB_1230_1240_Cobra_v1.23_DKBSCSIRom.bin

This might be a holistic thing, wherein whatever expansion is selected, the appropriate rom file is auto-added to the Expansion panel widgets, and saved appropriately to a saved config.uae file....

...the caveat here will be "if the required rom image is present in the ROM path" .... if it's not, the field should read "Not Found!"

General: rom dropdown widget really needs a ROM: label ...for the sake of clarity ;)

edit: and yes, the highlight around the scsi enable checkbox widget is incorrect...but maybe after guisan update I'll look again...

**Comment by giantclambake** [comment]
....had a bit of a read about this...

![ex3](https://imgur.com/9IaNpul.png)

....it would seem the checkbox outline appears like that, because the checkbox widget is expecting there to be a caption/label? I believe I have that correct ...(as if it wasn't already obvious ~ every other checkbox has a caption/label ;)

Either this got overlooked, or, the text of the caption/label becomes polemical ~ unlike other checkboxes, it refers to the selection of the dropdown to the left, which may change depending on what you're configuring...?...(other checkbox actions, refer directly to their respective caption/label, which is ostensibly 'fixed' )....

...so the text has to be neutral ... geez... how does one do this 'as briefly' as possible.... 'Enable option' ...??....'Option On/Off'... the 'correct' wording would be 'Check to enable' .....hmmm....

...either way, looks goofy as is... could be better  ;)

**Comment by giantclambake** [comment]
...well, I've tested all the available accelerator expansions, and they would appear to work...however...

*Expansions panel -> Accelerator Board Settings .... is 'thick as a brick' ....for example, several accelerator cards listed, IRL only appeared on say an A1200 ...ie; having an A3000 machine type, shouldn't see a DKB 1230/1240 being a possible expansion (should be the wildfire card instead), however it seems all accelerator expansions, can be add to -any- Amiga machine type....(and that's just wrong B)...

*I cannot discern any reason to reneg on what I said in #1365

**Reply by midwan** [reply]
That's exactly the same in WinUAE - either by mistake or because Toni didn't think it matters :D

**Reply by giantclambake** [reply]
Yeah, I noticed that ~ I've been checking against WinUAE ... it doesn't matter of course, accelerator emulation is like mascara ... B)
